### DISCUSSION TITLE
C++ Build Error while trying to run MuJoCo Code

### DISCUSSION THREAD

**Discussion by efecanalbay** [google-deepmind/mujoco#1970] [Asking for Help]
Hi,

I'm a student and I'm trying to use MuJoCo for observing and doing some measurements of walking pattern by using lower body of humanoid.

I'm looking for some help with running C++ code. I downloaded the zip file from https://github.com/google-deepmind/mujoco and try to run the code to see the simulation screen of MuJoCo. I already did this with Python or also did drag and drop easily. But I can not build my code in C++ right now. The reason behind that I need to run my code in C++ is I will get some measurement data as output and maybe extract some plots.

Here is a screenshot illustrating my question:

![Code_error](https://github.com/google-deepmind/mujoco/assets/171351827/854ea7c8-9c4e-46c9-9789-a2fdef7f0c9c)

Here is the written version of the error:

Starting build...
cmd /c chcp 65001>nul && C:\mingw64\bin\g++.exe -fdiagnostics-color=always simulate main.cc -L C:\Users\user\Desktop\mujoco-3.1.6-windows-x86_64\lib -g -lmujoco -std=c++20 C:\Users\user\Desktop\mujoco-3.1.6-windows-x86_64\simulate\main.cc -o C:\Users\user\Desktop\mujoco-3.1.6-windows-x86_64\simulate\main.exe
C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find simulate: No such file or directory
C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\user\AppData\Local\Temp\ccyRGBhE.o: in function `PhysicsThread(mujoco::Simulate*, char const*)':
C:/Users/user/Desktop/mujoco-3.1.6-windows-x86_64/simulate/main.cc:427: multiple definition of `PhysicsThread(mujoco::Simulate*, char const*)'; C:\Users\user\AppData\Local\Temp\cczp5o05.o:C:/Users/user/Desktop/mujoco-3.1.6-windows-x86_64/simulate/main.cc:427: first defined here
C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\user\AppData\Local\Temp\ccyRGBhE.o: in function `main':
C:/Users/user/Desktop/mujoco-3.1.6-windows-x86_64/simulate/main.cc:475: multiple definition of `main'; C:\Users\user\AppData\Local\Temp\cczp5o05.o:C:/Users/user/Desktop/mujoco-3.1.6-windows-x86_64/simulate/main.cc:475: first defined here
collect2.exe: error: ld returned 1 exit status

**Comment by dtorre38** [comment]
Did you build MuJoCo from source? If not, here are the instructions:

https://mujoco.readthedocs.io/en/stable/programming/index.html#building-from-source
